cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -442,9 +440,10 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, String namespac
     Map<NamespacedEntityId, Metadata> systemMetadata = fetchMetadata(sortedEntities, MetadataScope.SYSTEM);
     Map<NamespacedEntityId, Metadata> userMetadata = fetchMetadata(sortedEntities, MetadataScope.USER);
 
+    Set<MetadataSearchResultRecord> resultRecords = addMetadataToEntities(sortedEntities, systemMetadata, userMetadata);
     return new MetadataSearchResponse(
-      sortInfo.getSortBy() + " " + sortInfo.getSortOrder(), offset, limit, numCursors, total,
-      addMetadataToEntities(sortedEntities, systemMetadata, userMetadata), cursors, showHidden
+      sortInfo.getSortBy() + " " + sortInfo.getSortOrder(), offset, limit, numCursors, resultRecords.size(),
```

Comment:
Originally, it `total` would be 100. In this change, the number will now be `10`. I'm not sure if this is the "correct" behaviour but this is what the JIRA requested.

If this is incorrect, I can close the JIRA as won't fix.